src/persist-client/src/lib.rs

```diff
@@ -228,6 +228,32 @@ impl ShardId {
     }
 }
 
+/// Additional diagnostic information used within Persist
+/// e.g. for logging, metric labels, etc.
+#[derive(Clone, Debug)]
+pub struct Diagnostics {
+    shard_name: String,
+    handle_purpose: String,
+}
+
+impl Diagnostics {
+    /// Create a new `Diagnostics`.
+    pub fn new(shard_name: &str, handle_purpose: &str) -> Self {
+        Self {
+            shard_name: shard_name.to_string(),
+            handle_purpose: handle_purpose.to_string(),
+        }
+    }
+
+    #[cfg(test)]
+    pub(crate) fn for_tests() -> Self {
+        Self {
+            shard_name: "test".to_string(),
```

Comment:
nit: let's make these two strings different in some way (no strong opinion on the specifics)